src/arch/x86/reg/core64.rs

```diff
@@ -0,0 +1,148 @@
+use crate::arch::Registers;
+use crate::arch::x86::reg::F80;
+use core::convert::TryInto;
+
+/// 64-bit x86 core registers.
+///
+/// Source: https://github.com/bminor/binutils-gdb/blob/master/gdb/features/i386/64bit-core.xml
+#[derive(Default)]
+pub struct X86_64CoreRegs {
+    /// RAX, RBX, RCX, RDX, RSI, RDI, RBP, RSP, r8-r15
+    pub regs: [u64; 16],
+    /// Status register
+    pub eflags: u32,
+    /// Instruction pointer
+    pub rip: u64,
+    /// Segment registers: CS, SS, DS, ES, FS, GS
+    pub segments: [u32; 6],
+    /// FPU registers: ST0 through ST7
+    pub st: [F80; 8],
+    /// Floating-point control register
+    pub fctrl: u32,
```

Comment:
Thanks for looking into these floating-point registers!

Since they don't seem to be used very often nowadays, instead of cluttering up the main registers struct, lets extract these registers into a separate `struct X87FPUControlRegs` register. Feel free to tweak the naming if you like.